File: backend/candidate_app/services/candidate_education_service.py
from candidate_app.repositories.candidate_profile_repository import CandidateProfileRepository
from candidate_app.repositories.candidate_education_repository import CandidateEducationRepository

from rest_framework.exceptions import  NotFound


class CandidateEducationService:

    @staticmethod
    def create_education(user, validated_data):

        profile = CandidateProfileRepository.get_by_user(user)

        if not profile:
            raise NotFound("Candidate profile not found.")

        return CandidateEducationRepository.create(
            profile,
            validated_data
        )

    @staticmethod
    def get_all_educations(user):

        profile = CandidateProfileRepository.get_by_user(user)

        if not profile:
            raise NotFound("Candidate profile not found.")

        return CandidateEducationRepository.get_all_by_candidate(
            profile
        )

    @staticmethod
    def get_education(user, education_id):

        profile = CandidateProfileRepository.get_by_user(user)

        if not profile:
            raise NotFound("Candidate profile not found.")

        education = CandidateEducationRepository.get_by_id(
            education_id,
            profile
        )

        if not education:
            raise NotFound("Education not found.")

        return education

    @staticmethod
    def update_education(user, education_id, validated_data):

        profile = CandidateProfileRepository.get_by_user(user)

        if not profile:
            raise NotFound("Candidate profile not found.")

        education = CandidateEducationRepository.get_by_id(
            education_id,
            profile
        )

        if not education:
            raise NotFound("Education not found.")

        return CandidateEducationRepository.update(
            education,
            validated_data
        )

    @staticmethod
    def delete_education(user, education_id):

        profile = CandidateProfileRepository.get_by_user(user)

        if not profile:
            raise NotFound("Candidate profile not found.")

        education = CandidateEducationRepository.get_by_id(
            education_id,
            profile
        )

        if not education:
            raise NotFound("Education not found.")

        CandidateEducationRepository.delete(
            education
        ) 

File: frontend/src/services/candidate/candidateEducationService.js
import api from "../api";

export const createEducation = async (data) => {
  const response = await api.post(
    "/candidate/educations/",
    data
  );

  return response.data;
};

export const getEducations = async () => {
  const response = await api.get(
    "/candidate/educations/"
  );

  return response.data;
};

export const updateEducation = async (
  educationId,
  data
) => {
  const response = await api.put(
    `/candidate/educations/${educationId}/`,
    data
  );

  return response.data;
};

export const deleteEducation = async (
  educationId
) => {
  const response = await api.delete(
    `/candidate/educations/${educationId}/`
  );

  return response.data;
};